lib/Onnxifi/Base.cpp

```diff
@@ -26,6 +26,16 @@ bool BackendId::isOpSupported(Kinded::Kind opKind, ElemKind elementTy) {
   return executionEngine_.isOpSupported(opKind, elementTy);
 }
 
+void Backend::runAsync(const std::function<void(void)> &fn) {
+  auto &threadPool = getThreadPool();
+  threadPool.submit(fn);
+}
+
+ThreadPool &Backend::getThreadPool() {
+  static ThreadPool threadPool;
```

Comment:
we'd need thread pool per backend, not a single pool. Say you want to handle accelerator and CPU in the same time, for CPU you'll have a thread pool with hardware::concurrency number of threads and likely 1 thread for the accelerator.